src/utils/iconutils.cpp

```diff
@@ -75,6 +77,15 @@ QString IconUtils::replaceForegroundOfIcon(const QString &p_iconContent, const Q
     return p_iconContent;
 }
 
+bool IconUtils::isMonochrome(const QString &p_iconContent)
+{
+    QRegExp monoRe("#000000");
```

Comment:
You need to search the whole content to fetch all the `#xxxxxx` and check if all the colors are the same. If they are the same, then it is mono; otherwise, it is colorful.